cli/parse_args.ts

```diff
@@ -450,316 +473,250 @@ export function parseArgs<
-    collect?: boolean,
+    value: string | number | boolean,
+    arg: string,
+    collect: boolean,
   ) {
-    if (arg && flags.unknownFn && !argDefined(key, arg)) {
-      if (flags.unknownFn(arg, key, val) === false) return;
+    if (
+      !booleanSet.has(key) &&
+      !stringSet.has(key) &&
+      !aliasMap.has(key) &&
+      !(allBools && /^--[^=]+$/.test(arg)) &&
+      unknownFn?.(arg, key, value) === false
+    ) {
+      return;
     }
-
-    const value = !get(flags.strings, key) && isNumber(val) ? Number(val) : val;
-    setKey(argv, key, value, collect);
-
-    const alias = get(aliases, key);
-    if (alias) {
-      for (const x of alias) {
-        setKey(argv, x, value, collect);
-      }
+    if (typeof value === "string" && !stringSet.has(key)) {
+      value = isNumberString(value) ? Number(value) : value;
     }
-  }
 
-  function aliasIsBoolean(key: string): boolean {
-    return getForce(aliases, key).some(
-      (x) => typeof get(flags.bools, x) === "boolean",
+    const collectable = collect && collectSet.has(key);
+    setNested(argv, key.split("."), value, collectable);
+    aliasMap.get(key)?.forEach((key) =>
+      setNested(argv, key.split("."), value, collectable)
     );
   }
 
   let notFlags: string[] = [];
 
   // all args after "--" are not parsed
-  if (args.includes("--")) {
-    notFlags = args.slice(args.indexOf("--") + 1);
-    args = args.slice(0, args.indexOf("--"));
+  const index = args.indexOf("--");
+  if (index !== -1) {
+    notFlags = args.slice(index + 1);
+    args = args.slice(0, index);
   }
 
   for (let i = 0; i < args.length; i++) {
     const arg = args[i];
-    assertExists(arg);
-
-    if (/^--.+=/.test(arg)) {
-      const m = arg.match(/^--([^=]+)=(.*)$/s);
-      assertExists(m);
-      const [, key, value] = m;
-      assertExists(key);
-
-      if (flags.bools[key]) {
-        const booleanValue = value !== "false";
-        setArg(key, booleanValue, arg);
-      } else {
-        setArg(key, value, arg);
-      }
-    } else if (
-      /^--no-.+/.test(arg) && get(flags.negatable, arg.replace(/^--no-/, ""))
-    ) {
-      const m = arg.match(/^--no-(.+)/);
-      assertExists(m);
-      assertExists(m[1]);
-      setArg(m[1], false, arg, false);
-    } else if (/^--.+/.test(arg)) {
-      const m = arg.match(/^--(.+)/);
-      assertExists(m);
-      assertExists(m[1]);
-      const [, key] = m;
-      const next = args[i + 1];
-      if (
-        next !== undefined &&
-        !/^-/.test(next) &&
-        !get(flags.bools, key) &&
-        !flags.allBools &&
-        (get(aliases, key) ? !aliasIsBoolean(key) : true)
-      ) {
-        setArg(key, next, arg);
-        i++;
-      } else if (next !== undefined && (next === "true" || next === "false")) {
-        setArg(key, next === "true", arg);
-        i++;
-      } else {
-        setArg(key, get(flags.strings, key) ? "" : true, arg);
+
+    const groups = arg.match(FLAG_REGEXP)?.groups;
+
+    if (groups) {
```

Comment:
I tried to get rid of as many `else` clauses as possible and use early return or loop break or continuation instead.